internal/compiler/passes/lower_menus.rs

```diff
@@ -152,18 +152,27 @@ pub async fn lower_menus(
     assert!(matches!(&useful_menu_component.menu_entry, Type::Struct(..)));
 
     let mut has_menu = false;
+    let mut has_menubar = false;
 
     doc.visit_all_used_components(|component| {
         recurse_elem_including_sub_components_no_borrow(component, &(), &mut |elem, _| {
             if matches!(&elem.borrow().builtin_type(), Some(b) if b.name == "Window") {
-                has_menu |= process_window(elem, &useful_menu_component, diag);
+                has_menubar |= process_window(elem, &useful_menu_component, type_loader.compiler_config.no_native_menu, diag);
             }
             if matches!(&elem.borrow().builtin_type(), Some(b) if matches!(b.name.as_str(), "ContextMenuArea" | "ContextMenuInternal")) {
                 has_menu |= process_context_menu(elem, &useful_menu_component, diag);
             }
         })
     });
 
+    if has_menubar {
```

Comment:
This was not required before because the recurse_elem_including_sub_components_no_borrow a few line up was already visiting the menu ONLY IF, the menu was in a repeated component. And when not using native, since the `if BuiltinFunction::SupportsNativeMenuBar` is removed, it would just be a normal sub component which is not visited by the function earlier.